lib/components/map/connected-transitive-overlay.tsx

```diff
@@ -0,0 +1,94 @@
+import { Company, Itinerary, Leg } from '@opentripplanner/types'
+import { connect } from 'react-redux'
+import { useIntl } from 'react-intl'
+import React from 'react'
+import TransitiveCanvasOverlay, {
+  itineraryToTransitive
+  // @ts-expect-error transitive-overlay is not typescripted
+} from '@opentripplanner/transitive-overlay'
+
+import { getTransitiveData } from '../../util/state'
+
+type ItineraryData = {
+  companies?: Company[]
+  disableFlexArc: boolean
+  getTransitiveRouteLabel: (leg: Leg) => string
+  hasItineraryResponse: boolean
+  hasResponse: boolean
+  itineraryToRender: Itinerary
+  otpResponse: unknown
+}
+
+type Props = {
+  itineraryData: ItineraryData
+  labeledModes: string[]
+  styles: {
+    labels: Record<string, unknown>
+    segmentLabels: Record<string, unknown>
+  }
+}
+
+const ConnectedTransitiveOverlay = (props: Props) => {
+  const { itineraryData, labeledModes, styles } = props
+  const intl = useIntl()
+  if (!itineraryData) return null
+  const {
+    companies,
+    disableFlexArc,
+    getTransitiveRouteLabel,
+    hasItineraryResponse,
+    hasResponse,
+    itineraryToRender,
+    otpResponse
+  } = itineraryData
+
+  let transitiveData
+  if (hasResponse) {
+    if (hasItineraryResponse) {
+      transitiveData = itineraryToRender
+        ? itineraryToTransitive(itineraryToRender, {
+            companies,
+            disableFlexArc,
+            getTransitiveRouteLabel,
+            intl
+          })
+        : null
+    } else if (otpResponse) {
+      transitiveData = otpResponse
+    }
+  }
```

Comment:
Could this be moved to `mapStateToProps`? (e.g. by using `injectIntl` instead of `useIntl`, so that `intl` becomes an `ownProp`?) - This is to avoid repeated calculations of the transitive content on each render.

Reply:
Alternately, use the old code from file `state.js` but add an `intl` argument from `injectIntl` to `getTransitiveData` that you pass to `itineraryToTransitive`.

Reply:
This is some good thinking, thank you for that! I will refactor it.